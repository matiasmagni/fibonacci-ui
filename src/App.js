import logo from './logo.svg';
import './App.css';
import { useState } from 'react';
import axios from 'axios';

const baseUrl = 'http://localhost:3000';

const App = () => {

  const [num, setNum] = useState(0);
  const [fibonacciResult, setFibonacciResult] = useState(undefined);

  const handleChange = (event) => {
    setNum(event.target.value);
  };

  const handleSubmit = (event) => {
    event.preventDefault();

    axios.get(baseUrl + '/fibonacci?number=' + num)
      .then((response) => {
        // handle success
        setFibonacciResult(response.data.result);
      })
      .catch((error) => {
        // handle error
        console.log(error);
      })
  };

  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <p>
          Edit <code>src/App.js</code> and save to reload.
        </p>
        <form onSubmit={handleSubmit}>
          <label>
            Ingrese un n√∫mero para calcular la secuencia de Fibonacci:
            <input data-testid="num" id="num" type="num" name="num" value={num} onChange={handleChange} />
          </label>
          <input data-testid="calc" id="submitFibonacci" type="submit" value="Calcular" />
          {fibonacciResult && <p data-testid="result">Resultado: {fibonacciResult}</p>}
        </form>

      </header>
    </div>
  );
}

export default App;
