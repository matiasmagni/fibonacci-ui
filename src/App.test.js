import { fireEvent, render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import App from './App';

describe('App component', () => {
  it('App renders correctly!', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('App shows result after submitting a number.', async () => {
    const { getByTestId, findByTestId } = render(<App />);
    fireEvent.change(getByTestId('num'), { target: { value: 10 } });
    fireEvent.click(getByTestId('calc'));
    expect(await findByTestId('result')).toBeInTheDocument();
    expect(await findByTestId('result')).toHaveTextContent('Resultado: 55');
  });
});
