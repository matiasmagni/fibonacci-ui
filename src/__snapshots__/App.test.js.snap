// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component App renders correctly! 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <p>
      Edit 
      <code>
        src/App.js
      </code>
       and save to reload.
    </p>
    <form
      onSubmit={[Function]}
    >
      <label>
        Ingrese un n√∫mero para calcular la secuencia de Fibonacci:
        <input
          data-testid="num"
          id="num"
          name="num"
          onChange={[Function]}
          type="num"
          value={0}
        />
      </label>
      <input
        data-testid="calc"
        id="submitFibonacci"
        type="submit"
        value="Calcular"
      />
    </form>
  </header>
</div>
`;
